src/DataMatrix.java

```diff
@@ -1,133 +1,159 @@
          text = "";
    }
-   
-   //Accessors - Bryce
-   public int getActualWidth()
-   {
+
+   // Accessors - Bryce
+   public int getActualWidth() {
       return actualWidth;
    }
-   
-   public int getActualHeight()
-   {
+
+   public int getActualHeight() {
       return actualHeight;
    }
-   
-   //Jeannie
-   public boolean scan(BarcodeImage bc)
-   {
+
+   // Jeannie
+   public boolean scan(BarcodeImage bc) {
       // TODO Auto-generated method stub
       return false;
    }
 
+   // passing a String parameter to readText method.
+   // returning false if text is null
+   public boolean readText(String text) {
+      if (text == null)
+         return false;
 
- //Takes in a string variable and sets the data member "text"
-   public boolean readText(String text)
-   {
-   this.text = text;
-   return true;
+      this.text = text;
+      return true;
    }
 
-   //Jeannie
-   public boolean generateImageFromText()
-   {
+   // Jeannie
+   public boolean generateImageFromText() {
       // TODO Auto-generated method stub
       return false;
    }
 
-   //Harsandeep
-   public boolean translateImageToText()
-   {
-	   char colChar[] = new char[actualWidth - 1];
-	   for (int i = 1; i < actualWidth - 1; i++)
-	   {
-		   colChar[i] = this.readCharFromCol(i);
-	   }
-	   String colString = new String(colChar);
-	   this.text = colString;
-	   return true;
-	  }
-
-   //Jeannie
-   public void displayTextToConsole()
-   {
+// reading char from each column in image
+   private char readCharFromCol(int col) {
+      int readChar = 0;
+      int fromCol = 0;
+
+      for (int i = BarcodeImage.MAX_HEIGHT - 2; i > BarcodeImage.MAX_HEIGHT - actualHeight; i--) {
+         if (this.image.getPixel(i, col)) {
+            readChar = readChar + (int) Math.pow(2, fromCol);
+         }
+         fromCol++;
+      }
+      return (char) (readChar);
+   }
+
+// translating image to text by decoding each column
+   public boolean translateImageToText() {
+      char colChar[] = new char[actualWidth - 1];
+      for (int i = 1; i < actualWidth - 1; i++) {
+         colChar[i] = readCharFromCol(i);
+      }
+      String colString = new String(colChar);
+      text = colString;
+      return true;
+   }
+
+   // Jeannie
+   public void displayTextToConsole() {
       // TODO Auto-generated method stub
 
    }
 
-   //Bryce
-   public void displayImageToConsole()
-   {
+   // Bryce
+   public void displayImageToConsole() {
       // TODO Auto-generated method stub
 
    }
-   
-   //Harsandeep
-   private void cleanImage()
-   {
-	   for (int i = 0; i < BarcodeImage.MAX_HEIGHT; i++)
-	   {
-		   for (int j = 0; j < BarcodeImage.MAX_WIDTH; j++)
-		   {
-			   image.setPixel(i, j, false);
-		   }
-	   } 
+
+   // move the signal to the lower-left of the larger 2D array 
+   //with cleanImage method.
+   private void cleanImage() {
+      int startRow = 0;
+      int startColumn = 0;
+      boolean spine = false;
+
+      while (!spine) {
```

Comment:
This while loop seems unnecessary. Plus, it will become an infinite loop if image is empty.

Reply:
should I get rid of this while loop?